project_task_code/hooks.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2016 Tecnativa <vicent.cubells@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import SUPERUSER_ID, api
+
+
+def pre_init_hook(cr):
+    """
+    With this pre-init-hook we want to avoid error when creating the UNIQUE
+    code constraint when the module is installed and before the post-init-hook
+    is launched.
+    """
+    cr.execute("ALTER TABLE project_task ADD COLUMN code character varying;")
+    cr.execute("UPDATE project_task SET code = id;")
```

Comment:
Should be done only when the task code is activated? It's deactivated by default.
Same on `post_init_hook`

Reply:
Removed, not needed because it's deactivated by default.